# RNA-seq visualization

## Overview

This handout illustrates the introduction for RNA-sequencing and cover detailed information on three related visualization methods. 

## RNA-seq Visualization
The advancement in RNA sequencing revolutionized biologists' understanding in eukaryotes' transcriptome. It offers unprecedentedly precise measurement on the level of transcripts expression and their isoforms. Supplied with the tremendous amount of data generated by the Next Generation Sequencing, scientists require tools to display the data in a more explicit and straight-forward way. Subsequently, an appropriate and effective visualization tool plays critical role in data interpretation. This handout elaborates three commonly used method of visualization in RNA-seq: 
1. The Genome Alignment Graph
2. Principal Component Analysis Plot 
3. Heat Map

## 1. The Genome Alignment Graph
### Tools to Generate Genome Alignment Graph:
- Integrative Genomics Viewer(IGV)
- UCSC Genome Browser
The graph from this handout would use visualization produced by IGV as the example for illustration.
### Purpose and Instruction (from IGV official website): 
The Genome Alignment graph is a typical visualization widely used to help researchers analyze data from sequencing techniques such as Chip-seq and RNA-seq. It allows researchers to visualize how the genome from their samples aligns with the reference genome. 
- **Input Data**: Aligned reads from sequencing with format using BAM, SAM, or CRAM. IGV requires index files from BAM (with *bai* extension) and CRAM (with *crai* extension ) files.
- **Coverage Track**: Displays the depth of the reads as bar graph.
- **Alignment Track**: Displays the alignment details of the reads. This track provides information including insertions and deletions. 
- **Splice Junction Track**: Show the break of read coverage due to splicing. 
<img width="800" alt="IGV2" src="https://user-images.githubusercontent.com/83438583/205523473-177ed52a-d6fd-40f9-9437-6efa285b206f.png">
    
**Figure 1.** This image depicts IGV screenshot for genome sequence of human liver cell tissues. 

- **Viewing Alternative Splicing**: 
 <img width="800" alt="IGV1" src="https://user-images.githubusercontent.com/83438583/205523738-d9d3e3f3-efd6-4f2c-8a5a-8b3b7bd557ec.png">
 
**Figure 2.** It shows IGV screenshot for genome sequence of human liver cell and heart cell tissue.

 The graph provides an example of using IGV to observe alternative splicing. Human g18 is used as the reference genome, and the input files are the alignment files from human liver cell tissues and heart cell tissues. From the Coverage track and the Alignment track, it can be observed the sequence fragments align significantly differently in the gene *SLC25A3*, a phosphate transfer gene. This is one example of using IGV to monitor alternative splicing.   

## 2. Principal Component Analysis (PCA)
To simplify large genomic data such as the transcriptomes, Principal Component Analysis (PCA) is a useful classical dimension reduction approach that is often used to reduce the dimensionality of large data sets, by transforming a large set of variables into a smaller one while still capturing most of the information and variation. The reduction of large data sets is achieved by transforming the variables the principal components (PCs), which are orthogonal to each other. It is ordered such that the retention of variation present in the original variables decreases as moving down in the order. Therefore, most variation in the dataset would be along the first PC. After controlling for PC1, most remaining variation would be along the second PC. 

 <img width="800" alt="Screen Shot 2022-12-06 at 12 20 26 AM" src="https://user-images.githubusercontent.com/72646231/206057696-2592bfb5-48d1-477e-bed6-786008c842dc.png">
 
 **Figure 3.** Since PC1 and PC2 are perpendicular to each other, rotation is needed to make them straight so they can be the axes to construct a PCA plot.
 
 ### Purpose and Instruction
The purpose of PCA is to reduce dimensionality of large data sets while preserving as much variation as possible, making it possible to see strong patterns. Though it is not a clustering algorithm itself, the PCA plot often provides visually clustering for data exploration, which would be critical for differential expression analysis. It entails if the dataset has any outliers, any batch effect strongly influencing the dataset, or which samples are similar to one another, or which ones are different. Furthermore, by looking at the PCA loadings, which are the coefficients of the linear combination of the original variables from which the PCs are formed, researchers can identify the genes that are responsible for such similarities and differences. Since the first two PCs explain the most variation, so plotting the first 2 PCs would be essential in initial exploration of the dataset to determine the main driver for transcriptomic changes. 
 
 - **Tools and Packages**: R's prcomp() function and ggplot2 package to generate PCA plot and there are more tools and packages
 - **Input Data**: normalized data that follows a gaussian/binomial distribution. Since RNA-seq data is negative binomial distribution, log normalized data would be needed.

### Interpretation of the PCA Plot
 
As PCA is a way to bring out strong patterns from large and complex datasets, it can show cluster similarity. The samples with similar gene expression profiles will cluster together. 

![Blog_pca_8](https://user-images.githubusercontent.com/72646231/206060115-817dc790-4b75-4223-9f41-ffb394082234.png)
 
**Figure 4.** This image shows a PCA plot of cluster similarity such that there are 3 clusters. The first PC (explains most variation): Group 3 vs. Group 1 vs. Group 2&4. Though Group 2 and 4 seem to be undistinguishable using PC1 and PC2, they might be distinguishable when exploring other PC axis. 


  
### Application of the PCA Plot in Differential Expression Analysis

As PCA is critical in determining the main driver for transcriptomic changes in RNA-seq data, it provides useful information for researchers to decide the appropriate model for differential expression analysis. Below is a PCA plot of the first two PCs on 175 RNA-seq samples obtained from 20 patients with prostate cancer who underwent androgen deprivation therapy (ADT). The PCA plot clearly shows distinct clusters between pre-ADT and post-ADT as well as distinct clusters on methods by which the samples were obtained from patients. Such information is essential for choosing the model for differential expression analysis. If interested in genes that are differentially expressed between pre-ADT and post ADT and if there is a clear variation between the methods by which the samples were obtained from patients, researchers can potentially divide the dataset by the different methods and run differential expression analysis separately,or can use the method type as a fixed effect (covariate). 

<img width="800" alt="Screen Shot 2022-12-06 at 4 01 16 PM" src="https://user-images.githubusercontent.com/72646231/206061393-6bbdae85-1c4f-4132-ac87-043c1a9cfddf.png">

**Figure 5.** This image depicts a PCA plot of the first two PCs on 175 RNA-seq samples obtained from 20 patients with prostate cancer who underwent androgen deprivation therapy (ADT). RNA-seq samples include pre-ADT biopsies and post-ADT prostatectomy specimens.

 ## 3 Heat Map
 Heat map is a commonly used tool for data visualization, which uses the intensity of color to show the magnitude of a phenomenon in different samples. In RNA-seq, heat map is used to compare to transcripts expression profile across different specimen. 

 ### How to Generate a Heat Map

 - **Tools:** 
 Heatmap function in ggplot package in R can be used as a tool to generate heat map, but there are also other packages available.

 - **Input:**
 To generate a heat map, we need to input the differential expression file after the differential analysis step. Each data point in a heat map is a column of 3 elements, top differentially expressed genes, sample names, and normalized read counts of transcripts. It is usually displayed as a table.

|       | Sample 1 | Sample 2 | Sample 3 | Sample 4 |
|------:|---------:|---------:|---------:|---------:|
|Gene 1 | 2350     | 1038     | 3102     | 1302     |
|Gene 2 | 5340     | 1039     | 6102     | 1012     |
|Gene 3 | 328      | 1226     | 354      | 1289     |

**Table 1** It is an example of a differential expression file. The numbers represent the normalized read counts.

 ### Data Interpretation 

Each rectangle in a heat map represents a data point. The color measures the expression level of the genes. In the example below, red color represents that the gene is upregulated in the sample compared to the other sample, and the blue means the opposite. 

 <img width = '600'  alt = 'heatmap' src = 'https://media.springernature.com/full/springer-static/image/art%3A10.1186%2F1471-2180-14-31/MediaObjects/12866_2013_Article_2194_Fig7_HTML.jpg?as=webp'>

 **Figure 6.** An example of a heat map generated by RNA-seq data. This heat map is after clustering. 

 The intensity of the color reflects the magnitude of the differential expression. For example, a deeper red color means that this gene is even more upregulated in the sample compared to the samples with lighter red. 

 ### Further Steps 

After we have generated the heat map, it is common to cluster the samples or the genes by non-supervised machine learning. For example, in Figure 6, By clustering the sample with the similar expression pattern, we can clearly see how the samples are different form each other in terms of the gene expression.  


## References
1. https://software.broadinstitute.org/software/igv/AlignmentData
2. https://www.google.com/search?q=igv+rna+seq&rlz=1C5CHFA_enUS983US984&biw=1336&bih=1013&tbm=vid&ei=yiyNY5DjAfelqtsPl8CugAY&ved=0ahUKEwiQ8LbFjuH7AhX3kmoFHRegC2AQ4dUDCA0&uact=5&oq=igv+rna+seq&gs_lcp=Cg1nd3Mtd2l6LXZpZGVvEAMyBQgAEJECMgYIABAWEB4yBggAEBYQHjIICAAQFhAeEA8yCAgAEBYQHhAPMgYIABAWEB4yBQgAEIYDMgUIABCGAzoFCCEQoAE6BQgAEIAEOgQIABBDOgcIABCABBANOgYIABAeEA06CAgAEB4QDxANOgoIABAIEB4QDxANOggIABAIEB4QDVCWA1jnMWDrMmgOcAB4AIABbogBjA2SAQQxNi4ymAEAoAEBwAEB&sclient=gws-wiz-video#fpstate=ive&vld=cid:367671eb,vid:awGN-rpLYas
3. Chua, K.Y., Monk, I.R., Lin, YH. et al. Hyperexpression of α-hemolysin explains enhanced virulence of sequence type 93 community-associated methicillin-resistant Staphylococcus aureus . BMC Microbiol 14, 31 (2014). 
4. https://www.youtube.com/watch?v=dx9-N8b9Yj4
5. https://builtin.com/data-science/step-step-explanation-principal-component-analysis
6. https://www.projectpro.io/data-science-in-python-tutorial/principal-component-analysis-tutorial
7. https://alexslemonade.github.io/refinebio-examples/03-rnaseq/dimension-reduction_rnaseq_01_pca.html
8. https://blog.bioturing.com/2018/06/14/principal-component-analysis-explained-simply/
9. https://scentellegher.github.io/machine-learning/2020/01/27/pca-loadings-sklearn.html#:~:text=PCA%20loadings%20are%20the%20coefficients,components%20(PCs)%20are%20constructed.
10. Wang, S., Zhang, Y., Hu, C. et al. Shiny-DEG: A Web Application to Analyze and Visualize Differentially Expressed Genes in RNA-seq. Interdiscip Sci Comput Life Sci 12, 349–354 (2020). https://doi.org/10.1007/s12539-020-00383-7

